hs_core/views/__init__.py

```diff
@@ -1254,6 +1297,8 @@ def create_resource(request, *args, **kwargs):
         ajax_response_data['status'] = 'success'
         ajax_response_data['resource_url'] = resource.get_absolute_url()
 
+    web_services.update(resource.short_id)
```

Comment:
Need to put web_services.update() call into a celery task so it runs in a background thread so as not to block the main thread, especially given the calling functions are called very frequently.

Reply:
Also, these web_services.update() call does not cover REST API scenarios but only cover the web UI triggers. Suggest to put these web_services.update() call into common utility functions that are called by both these view functions and REST API calls so that web services can be updated in all scenarios when files are added/removed in HydroShare.

Reply:
yeah, I think we need to consider using events that web_services and others can hook into instead of answering a line even if it's offloaded to another thread.

Reply:
@sblack-usu yeah, agreed firing up events via signals are a better option for others to listen and hook into as needed. @kjlippold there are many events/signals already fired up for many actions in HydroShare, so you may be able to leverage existing signals or adding missing ones as needed to serve your web service needs. One example for your reference here: https://github.com/hydroshare/hydroshare/blob/develop/hs_core/hydroshare/resource.py#L1015

Reply:
@kjlippold yes, there is no signal for that. Here is the method for setting a resource public: https://github.com/hydroshare/hydroshare/blob/develop/hs_core/models.py#L1748, and the view function https://github.com/hydroshare/hydroshare/blob/562cfa5bd0d39722eaa4816b729056caeb101fc6/hs_core/views/__init__.py#L761